content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md

```diff
@@ -0,0 +1,276 @@
+---
+date: "2022-08-23"
+title: "What's new in axum 0.6.0-rc.1"
+description: "August 23, 2022"
+---
+
+Today, we're happy to announce [`axum`] version 0.6.0-rc.1. `axum` is an ergonomic and
+modular web framework built with `tokio`, `tower`, and `hyper`.
+
+In 0.6 we've reworked some fundamental parts of axum to make things more type
+safe, less surprising, and easier to use. In this post I'd like to highlight a
+few of the most impactful changes and new features.
+
+This also includes new major versions for [`axum-core`], [`axum-extra`], and
+[`axum-macros`].
+
+## Type safe `State` extractor
+
+Previously the recommended way to share state with handlers was to use the
+`Extension` middleware and extractor:
+
+```rust
+use axum::{
+    Router,
+    Extension,
+    routing::get,
+};
+
+#[derive(Clone)]
+struct AppState {}
+
+let state = AppState {};
+
+let app = Router::new()
+    .route("/", get(handler))
+    // Add `Extension` as a middleware
+    .layer(Extension(state));
+
+async fn handler(
+    // And extract our shared `AppState` using `Extension`
+    Extension(state): Extension<AppState>,
+) {}
+```
+
+However this wasn't type safe, so if you forgot `.layer(Extension(...))` things
+would compile just fine but you'd get runtime errors when calling `handler`.
+
+In 0.6 you can use the new `State` extractor which works similarly to
+`Extension` but is type safe:
```

Comment:
Type safe or statically checked? I don't feel like that is what type safe means right?

Reply:
Uhm whats the difference? 😅

I mean that if you attempt to extract the wrong state type then it wont compile.